engine/crates/engine-v2/src/engine.rs

```diff
@@ -285,6 +285,7 @@ impl<'ctx, R: Runtime> PreExecutionContext<'ctx, R> {
         let response = if matches!(operation.ty(), OperationType::Subscription) {
             Response::bad_request(GraphqlError::new(
                 "Subscriptions are only suported on streaming transports. Try making a request with SSE or WebSockets",
+                ErrorCode::BadRequest,
```

Comment:
If we already call `Response::bad_request`, could this automatically generate a `GraphqlError` with the corresponding error code?

Reply:
fair point

Reply:
Ok not the same thing, bad request here means it's a request error in the GraphQL-spec sense. So no `data` field. Error could also be validation one or persisted query not found.

Reply:
renamed to `pre_execution_error` which is closer to the spec meaning